apps/web/src/server/public-api/hono.ts

```diff
@@ -1,11 +1,54 @@
-import { OpenAPIHono } from "@hono/zod-openapi";
 import { swaggerUI } from "@hono/swagger-ui";
-import { handleError } from "./api-error";
+import { OpenAPIHono } from "@hono/zod-openapi";
 import { env } from "~/env";
+import { db } from "~/server/db";
+import { handleError } from "./api-error";
+import { getApiKeyFromToken, getTeamFromToken } from "./auth";
 
 export function getApp() {
   const app = new OpenAPIHono().basePath("/api");
 
+  // Add logging middleware
+  app.use("*", async (c, next) => {
+    const start = performance.now();
+
+    await next();
+
+    const end = performance.now();
+
+    const duration = Math.round(end - start);
+
+    try {
+      const team = await getTeamFromToken(c);
+      const apiKey = await getApiKeyFromToken(c);
```

Comment:
apiKey table as team id, so this will avoid 2 queries everytime

Reply:
I didn't get this. Can you explain a bit more?

Reply:
getTeamFromToken, getApiKeyFromToken makes 2 api queries and it's inefficient. we can update `getApiKeyFromToken ` such that it can return teamId as well

Reply:
Made new function getApiKeyAndTeamFromToken() which utilizes existing function